ast/src/main/java/com/graphicsfuzz/common/ast/decl/ArrayInfo.java

```diff
@@ -17,27 +17,95 @@
 package com.graphicsfuzz.common.ast.decl;
 
 import com.graphicsfuzz.common.ast.IAstNode;
+import com.graphicsfuzz.common.ast.expr.Expr;
+import com.graphicsfuzz.common.ast.expr.IntConstantExpr;
 import com.graphicsfuzz.common.ast.visitors.IAstVisitor;
+import com.graphicsfuzz.common.ast.visitors.UnsupportedLanguageFeatureException;
 import java.util.Optional;
 
 public class ArrayInfo implements IAstNode {
 
-  private Optional<Integer> size;
+  /**
+   * Size expression, should be IntConstantExpr after folding.
+   */
+  private Optional<Expr> constantSize;
+  /**
+   * Original size expression before folding, for prettyprint.
+   */
+  private Optional<Expr> originalSize;
 
-  public ArrayInfo(Integer size) {
-    this.size = Optional.of(size);
+  /**
+   * Constructor, stores the original expression and a clone for constant folding.
+   * @param size Array size expression
+   */
+  public ArrayInfo(Expr size) {
```

Comment:
I would not clone "size" here - we don't do that elsewhere in the code base, instead preferring to clone when we know we need to rather than by default - and I'd rather stay consistent.

Can you change the parameter name to "originalSize" assuming you make the previous change?

Reply:
I think I've done all of those.